### PR TITLE
Implement toy example for node classification with PyG

### PR DESCRIPTION
**Pull request by mdujic**
### Description

Please briefly explain the changes you made here.

### Pull request type

- [ ] Bugfix
- [ ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)

######################################

### CODE REVIEW

File: example/node_classification.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    model.eval()
+    out = model(data.x, data.edge_index)
+    pred = out.argmax(dim=1)  # Use the class with highest probability.
+    correct = pred[mask] == data.y[mask]  # Check against ground-truth labels.
+    acc = int(correct.sum()) / int(mask.sum())  # Derive ratio of correct predictions.
+    return acc
+
+def convert_data(nodes, edges):
+    if len(nodes) == 0:
+        return "Insert data in database first!"
```
Comment (antepusic) [inline example/node_classification.py:59]: Situations like this are better handled by throwing an exception.

File: example/node_classification.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
+    global criterion 
+    criterion = torch.nn.CrossEntropyLoss()
+    
+    for epoch in range(1, 101):
+        loss = train(data)
+        print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}')
+    
+    test_acc = test(data, data.test_mask)
+    print(f'Test Accuracy: {test_acc:.4f}')
```
Comment (antepusic) [inline example/node_classification.py:129]: While it probably doesn’t matter in a toy example like this one, **always** split the train and test sets.

File: example/node_classification.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    test_mask = torch.tensor(test_mask, dtype=torch.bool)
+
+    data = Data(x, y, edge_index, train_mask, val_mask, test_mask)
+    return data
+
+
+@mgp.read_proc
+def work(
+    ctx: mgp.ProcCtx
+) -> mgp.Record(no_nodes=int, no_edges=int):
```
Comment (antepusic) [inline example/node_classification.py:104]: Consider separating `train` and `test`/`evaluation` functionalities by having, for example, multiple procedures or multiple modes of operation in the module.

File: example/node_classification.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        self.train_mask = train_mask
+        self.val_mask = val_mask
+        self.test_mask = test_mask
+
+class GCN(torch.nn.Module):
+    def __init__(self, hidden_channels):
+        super().__init__()
+        torch.manual_seed(1234567)
+        self.conv1 = GCNConv(1433, hidden_channels)
+        self.conv2 = GCNConv(hidden_channels, 7)
```
[the remark this answers is not in the document]
Comment (mdujic) [reply example/node_classification.py:24]: Repaired.

File: example/node_classification.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    return loss
+
+
+
+def test(data, mask):
+    model.eval()
+    out = model(data.x, data.edge_index)
+    pred = out.argmax(dim=1)  # Use the class with highest probability.
+    correct = pred[mask] == data.y[mask]  # Check against ground-truth labels.
+    acc = int(correct.sum()) / int(mask.sum())  # Derive ratio of correct predictions.
```
[the remark this answers is not in the document]
Comment (mdujic) [reply example/node_classification.py:54]: Thank you for this one, I'll definitely have a look right now.

File: example/node_classification.py (hunk 1 above)
Comment (mdujic) [reply example/node_classification.py:59]: I was afraid of throwing exceptions, thank you for enlightment. I've repaired now. (in new branch)

File: example/node_classification.py (hunk 2 above)
Comment (mdujic) [reply example/node_classification.py:129]: For now, training, validation and test data are separated by train_mask, val_mask, test_mask. If i-th example is for training, it has train_mask[i] = 1 and val_mask[i] = 0, test_mask[i] = 0.

File: example/node_classification.py (hunk 3 above)
Comment (mdujic) [reply example/node_classification.py:104]: I've repaired that also, thanks!

**Pull request closed without merging**